src/Store/VimStoreConnector.re

```diff
@@ -294,6 +294,8 @@ let start =
         let fileType =
           switch (metadata.filePath) {
           | Some(v) =>
+            // TODO: Can't be changed, as buffer is only made a few lines below.
+            // But we probably could re-run once the buffer is made, in BufferEnter?
             Some(
               Exthost.LanguageInfo.getLanguageFromFilePath(languageInfo, v),
             )
```

Comment:
I believe we could move this to after we create the buffer, but before we `setFileType` - I think that would work